py/Boinc/setup_project.py

```diff
@@ -345,25 +345,29 @@ def dir(*dirs):
         except OSError:
             pass
         try:
-            s = subprocess.Popen(["git", "rev-parse", "HEAD"], stdout=subprocess.PIPE, stderr=open(os.devnull, 'w'))
-            commit = s.stdout.read()[:7]
-            s = subprocess.Popen(["git", "describe", "--match", "server_release/*", "--tags", "--exact", "HEAD"],
+            s = subprocess.Popen(["git", "config", "--get", "remote.origin.url"],
                                  stdout=subprocess.PIPE, stderr=open(os.devnull, 'w'))
-            version = s.stdout.read().split("/")[-1].strip()
-            content = '''<?php
-global $git_commit;
+            git_url = s.stdout.read().replace('.git','/tree/').strip()
+            s = subprocess.Popen(["git", "describe", "--match", "server_release/*"],
+                                 stdout=subprocess.PIPE, stderr=open(os.devnull, 'w'))
+            version = s.stdout.read().strip()
+            git_url += version
+            version = version.split("/")[-1]
+
+            if version:
+                content = '''<?php
 global $server_version;
-$git_commit = "{commit}";
+global $git_url;
 $server_version = "{version}";
-
+$git_url = "{git_url}";
 ?>
-'''.format(commit=commit, version=version)
-            f = open(os.path.join(dest_dir, 'release.inc'), 'w')
-            f.write(content)
-            f.close()
-            os.chmod(os.path.join(dest_dir, 'release.inc'), 0o644)
+'''.format(git_url=git_url, version=version)
+                f = open(os.path.join(dest_dir, 'release.inc'), 'w')
```

Comment:
Is this file only written once during the initial project instantiation? Shouldn't the version be a dynamic detail that always represents the current state?

Reply:
It is also created when doing upgrade.